qfieldsync/core/cloud_api.py

```diff
@@ -114,13 +122,73 @@ def from_reply(reply: QNetworkReply) -> Optional[CloudException]:
     return CloudException(reply, Exception(message))
 
 
+class CloudAuthMethod(Enum):
+    NONE = 0
+    CREDENTIALS = 1
+    SSO = 2
+
+
+def build_oauth2_auth_config_from_cloud_method(
+    auth_data: dict, related_uri: str, config_name: str = "qfieldcloud-sso"
+) -> QgsAuthMethodConfig:
+    """Builds a QgsAuthMethodConfig from a method provided by QFieldCloud's auth capabilities.
+
+    Args:
+        auth_data (dict): dict describing an auth method.
+        related_uri (str): URI of the generated auth config.
+        config_name (str, optional): name of the generated auth config. Defaults to "qfieldcloud-oauth2".
+
+    Returns:
+        QgsAuthMethodConfig: QGIS auth config for the provided method.
+    """
+    auth_config_dict = {
+        "method": "oauth2",
+        "accessMethod": 0,
+        "grantFlow": auth_data["grant_flow"],
+        "configType": 1,
+        "clientId": auth_data["client_id"],
+        "clientSecret": auth_data["client_secret"],
+        "extraTokens": auth_data["extra_tokens"],
+        "requestUrl": auth_data["request_url"],
+        "tokenUrl": auth_data["token_url"],
+        "redirectHost": auth_data["redirect_host"],
+        "redirectPort": auth_data["redirect_port"],
+        "redirectUrl": auth_data["redirect_url"],
+        "persistToken": False,
+        "pkceEnabled": auth_data["pkce_enabled"],
+        "refreshTokenUrl": auth_data["refresh_token_url"],
+        "requestTimeout": 30,
+        "version": 1,
+        "scope": auth_data["scope"],
+    }
+
+    auth_config_str = str(auth_config_dict)
+    auth_config_str = auth_config_str.replace("'", '"')
+    auth_config_str = auth_config_str.replace("False", "false")
+    auth_config_str = auth_config_str.replace("True", "true")
+    auth_config_map = {"oauth2config": auth_config_str}
+
+    auth_config = QgsAuthMethodConfig()
+    auth_config.setMethod("OAuth2")
+    auth_config.setVersion(1)
+    auth_config.setName(config_name)
+    auth_config.setUri(related_uri)
+    auth_config.setConfigMap(auth_config_map)
```

Comment:
I'd avoid using setConfigMap in favor of setConfig.

```suggestion
    auth_config.setConfig("oauth2config", auth_config_str)
```